sample-tuning-setup/rayjob.yaml

```
@@ -0,0 +1,103 @@
+            # resource accounting. K8s requests are not used by Ray.
+            resources:
+              limits:
+                cpu: "14"
+                memory: "54Gi"
+              requests:
+                cpu: "14"
+                memory: "54Gi"
+            env:
+            - name: RAY_OBJECT_STORE_MEMORY
+              value: "16G"
+            lifecycle:
+              preStop:
+                exec:
+                  command: ["/bin/sh","-c","ray stop"]
+            volumeMounts:
+            - name: pvc-blob-results
+              mountPath: /results
+          volumes:
+          - name: pvc-blob-results
+            persistentVolumeClaim:
+              claimName: pvc-blob-results
+    workerGroupSpecs:
+    - replicas: 9
+      minReplicas: 9
+      maxReplicas: 9
+      # To experiment with autoscaling,
+      # set replicas and minReplicas to 0.
+      # replicas: 0
+      # minReplicas: 0
+      groupName: large-group
+      # the following params are used to complete the ray start: ray start --block
+      rayStartParams:
+        block: 'true'
+      template:
+        spec:
+          containers:
+          - name: machine-learning # must consist of lower case alphanumeric characters or '-', and must start and end with an alphanumeric character (e.g. 'my-name',  or '123-abc')
+            image: rayproject/ray-ml:2.43.0.84f276-py310-cpu
+            # Optimal resource allocation will depend on your Kubernetes infrastructure and might
+            # require some experimentation.
+            # Setting requests=limits is recommended with Ray. K8s limits are used for Ray-internal
+            # resource accounting. K8s requests are not used by Ray.
+            resources:
+              limits:
+                # Slightly less than 16 to accomodate placement on 16 vCPU virtual machine.
+                cpu: "14"
+                memory: "54Gi"
+                # The node that hosts this pod should have at least 1000Gi disk space,
+                # for data set storage.
+              requests:
+                cpu: "14"
+                memory: "54Gi"
+            env:
+            - name: RAY_OBJECT_STORE_MEMORY
+              value: "16G"
+            lifecycle:
+              preStop:
+                exec:
+                  command: ["/bin/sh","-c","ray stop"]
+            volumeMounts:
+            - name: pvc-blob-results
+              mountPath: /results
+          volumes:
+          - name: pvc-blob-results
+            persistentVolumeClaim:
+              claimName: pvc-blob-results
+          initContainers:
+          # the env var $RAY_IP is set by the operator, with the value of the head service name
+          - name: init-myservice
+            image: busybox:1.28
+            command: ['sh', '-c', "until nslookup $RAY_IP.$(cat /var/run/secrets/kubernetes.io/serviceaccount/namespace).svc.cluster.local; do echo waiting for myservice; sleep 2; done"]
```

Can we add PVC config also (azure blob will dynamically provision resources)?
```suggestion
            command: ['sh', '-c', "until nslookup $RAY_IP.$(cat /var/run/secrets/kubernetes.io/serviceaccount/namespace).svc.cluster.local; do echo waiting for myservice; sleep 2; done"]
---
kind: PersistentVolumeClaim
apiVersion: v1
metadata:
  name: pvc-blob-results
spec:
  accessModes:
    - ReadWriteMany
  resources:
    requests:
      storage: 200Gi
  storageClassName: azureblob-fuse-premium
```

---

pvc.yaml is added here: https://github.com/Azure-Samples/aks-ray-sample/blob/main/sample-tuning-setup/pvc.yaml
No specific reason for using static provisioning, dynamic provisioning can also be used, this is just how I tested it.

---

@nearora-msft  do we want to provide a reference to the project for the driver? https://github.com/Azure/azure-storage-fuse or to the docs https://learn.microsoft.com/en-us/azure/aks/azure-blob-csi?tabs=NFS for SEO

---

@mittachaitu has another PR with some changes related to provisioning and PVC. Maybe he can add that in his changes?